partner_manual_rank/models/res_partner.py

```diff
@@ -7,19 +7,27 @@
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
+    def _default_is_customer(self):
+        return self.env.context.get("res_partner_search_mode") == "customer"
+
+    def _default_is_supplier(self):
+        return self.env.context.get("res_partner_search_mode") == "supplier"
+
     is_customer = fields.Boolean(
         compute="_compute_is_customer",
         inverse="_inverse_is_customer",
         store=True,
         readonly=False,
         string="Is a Customer",
+        default=_default_is_customer,
```

Comment:
Please don't harcode methods this way, because:
- It will make the method non-inheritable
- It will force methods to be above fields, which is against guidelines

```suggestion
        default=lambda self: self._default_is_customer(),
```

Same below